src/index.js

```diff
@@ -38,9 +38,15 @@ const dayjs = function (date, c) {
     return date.clone()
   }
   // eslint-disable-next-line no-nested-ternary
-  const cfg = typeof c === 'object' ? c : {}
+  let cfg = {}
+  if ((typeof c === 'object') && !Array.isArray(c)) {
+    // clone c (contains cfg object)
+    cfg = JSON.parse(JSON.stringify(c))
+  }
   cfg.date = date
-  cfg.args = arguments// eslint-disable-line prefer-rest-params
+  if (cfg.args === undefined) {
+    cfg.args = [...arguments] // eslint-disable-line prefer-rest-params
```

Comment:
it's better to avoid using the spread operator, cause it will increase the final compiled js bundle size dramatically.

Reply:
I replaced the spread operator with Array.prototype.slice.  
Searching for other files using the spread operator I found a few, but there the solution is not so simple, as here the spread operator is used to extend an object.

Perhaps it makes sense to think about what systems to support. I opened issue #1925 to discuss that topic.